### PR TITLE
[V2] Fix async filter call in combinators

### PR DESCRIPTION
**Pull request by Jahongir-Qurbonov**

### CODE REVIEW

**Comment by Jahongir-Qurbonov** [comment]
Or is it better to remove the awaitable check and make all filters only awaitable?

File: client/src/telethon/_impl/client/events/filters/combinators.py [hunk 1]
```diff
@@ -81,8 +82,10 @@ def filters(self) -> tuple[Filter, ...]:
         """
         return self._filters
 
-    def __call__(self, event: Event) -> bool:
-        return any(f(event) for f in self._filters)
+    async def __call__(self, event: Event) -> bool:
+        return any(
+            [await r if isawaitable(r := f(event)) else r for f in self._filters]
```
Comment (Lonami) [inline client/src/telethon/_impl/client/events/filters/combinators.py]: This is no longer lazy. It evaluates the entire list upfront.

If removing the `[]` doesn't work, it should be rewritten as a `for` loop.

File: client/src/telethon/_impl/client/events/filters/combinators.py [hunk 2]
```diff
@@ -121,8 +124,10 @@ def filters(self) -> tuple[Filter, ...]:
         """
         return self._filters
 
-    def __call__(self, event: Event) -> bool:
-        return all(f(event) for f in self._filters)
+    async def __call__(self, event: Event) -> bool:
+        return all(
+            [await r if isawaitable(r := f(event)) else r for f in self._filters]
```
Comment (Lonami) [inline client/src/telethon/_impl/client/events/filters/combinators.py]: Same here.

**Review by Lonami — CHANGES_REQUESTED** [review]

**Comment by Lonami** [comment]
> make all filters only awaitable

Creating filters with `lambda` would be far more annoying.

As long as the type checking is happy with maybe-optional, I'm fine with it supporting both, too.

**Pull request merged**